src/pcre2_compile_class.c

```diff
@@ -1844,11 +1845,19 @@ while (TRUE)
   /* Join second and subsequent leaves with an OR. */
   if (!first) *code++ = OP_ECLASS_OR;
 
+  if (lengthptr != NULL)
+    {
+    *lengthptr += code - code_start;
+    code = code_start;
+    }
+
   first = FALSE;
   }
 
 DONE:
 PCRE2_ASSERT(!first);  /* Confirm that we found something. */
+PCRE2_ASSERT(lengthptr == NULL ||
+             (code == code_start && *pcode == code_start));
```

Comment:
You added an extra condition to the assertion here - `*pcode == code_start`, which the other assertion doesn't have. I don't think it's needed, since there's literally no code in the function which touches `pcode`.